### PR TITLE
Fix and extend doc blocks and parameter hinting

### PR DESCRIPTION
**Pull request by distantnative**

### CODE REVIEW

File: src/Cms/Site.php [hunk 1]
```diff
@@ -360,11 +360,15 @@ public function mediaUrl(): string
      *
      * @param string|null $format
      * @param string|null $handler
-     * @return mixed
+     * @return int|string
```
Comment (afbora) [inline src/Cms/Site.php:363]: `Dir::modified()` returns `int`?

https://github.com/getkirby/kirby/blob/3.5.3.1/src/Toolkit/Dir.php#L257

Comment (lukasbestle) [reply src/Cms/Site.php:363]: That type hint is incorrect as well. If a format is used, the return value is indeed `string`.

**Review by lukasbestle — COMMENTED** [review]
I agree on all changes. However I think we should wait for 3.6.0 before we merge this as some type hints make the behavior stricter.

File: src/Cms/Site.php (hunk 1 above)
Comment (lukasbestle) [reply src/Cms/Site.php:363]: That type hint is incorrect as well. If a format is used, the return value is indeed `string`.

**Comment by distantnative** [comment]
@lukasbestle I will move all changes that make the behavior stricter to a separate PR. That way we can get the other doc fixes faster into 3.5.4 already.

**Review by lukasbestle — APPROVED** [review]

**Pull request merged**